### PR TITLE
[16.0] [IMP] fastapi: Factorize error handling and  use it in tests with raise_server_exceptions=False

### PR DESCRIPTION
**Pull request by paradoxxxzero**
As of now using `raise_server_exceptions=False` leads to all exceptions to be returned as 500.
 
This PR aims to share the error handling between odoo FastAPIDispatcher and the starlette TestClient to get meaningful response in tests.

This also adds a log of 500 exceptions in testing which is very useful to get unhandled exceptions traceback while running the tests.

### CODE REVIEW

File: fastapi/routers/demo_router.py [hunk 1]
```diff
@@ -4,6 +4,7 @@
 The demo router is a router that demonstrates how to use the fastapi
 integration with odoo.
 """
+
```
Comment (lmignon) [inline fastapi/routers/demo_router.py]: @paradoxxxzero Can you avoid to modify this file just for cosmetic?

**Review by lmignon — CHANGES_REQUESTED** [review]
Thank you @paradoxxxzero for this useful change. A little comment to reduce the diff and it'll be perfect.

**Comment by paradoxxxzero** [comment]
> Thank you @paradoxxxzero for this useful change. A little comment to reduce the diff and it'll be perfect.

Done :)

**Review by lmignon — APPROVED** [review]

**Comment by lmignon** [comment]
ping @AnizR

**Review by AnizR — APPROVED** [review]
Great! Thanks for your contribution.

Code LGTM

**Pull request merged**